### PR TITLE
Introduce terraform module test framework

### PR DESCRIPTION
**Pull request by JunyiYi**
* Introduce Gemfile and Rake
* Update documentation
* Update the terraform code to meet the format requirement
* Introduce DockerFile

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,194 @@
+GIT
```
Comment (metacpp) [inline Gemfile.lock]: This file don't need to be checked in.

**Comment by metacpp** [comment]
@JunyiYi you need to enable the Travis CI for it.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,53 @@ module "sql-database" {
 }
```
Comment (metacpp) [inline README.md]: Don't forget the badge.

**Comment by metacpp** [comment]
@dcaro , can you grant the admin permission to the terraform team ?

**Comment by dcaro** [comment]
@metacpp terraformteam has permissions.

**Comment by metacpp** [comment]
@dcaro I still can't see the "Settings" tab on this repo. Can you double check if "terraform-azure" is granted as admin not just the contributor only ?

File: README.md (hunk 2 above)
Comment (JunyiYi) [reply README.md]: Will do after travis is enabled.

**Pull request merged**